gradle/build.go

```diff
@@ -117,3 +149,37 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 
 	return result, nil
 }
+
+func handleGradleSettings(binding libcnb.Binding, gradleHome string, md map[string]interface{}) error {
+	path, ok := binding.SecretFilePath("gradle.properties")
+	if !ok {
+		return nil
+	}
+
+	gradleProperties, err := ioutil.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("error reading gradle.properties\n%w", err)
+	}
+
+	err = os.MkdirAll(gradleHome, 0755)
+	if err != nil {
+		return fmt.Errorf("cannot make gradle home\n%w", err)
+	}
+
+	err = ioutil.WriteFile(filepath.Join(gradleHome, "gradle.properties"), gradleProperties, 0644)
```

Comment:
`gradle.properties`can contain sensitive data so I don't think we can copy it into a layer like this. Imagining this build is in running in k8s writing `gradle.properties` here will create copies of the secret in two locations
1. The FS of the stopped container
2. A persistent volume (this is a cached layer and a TARed copy will likely be persisted and restored on the next build
Whatever access and RBAC policies are applied to k8s `Secrets`, they are likely to be different for persistent volume mounts and I don't know all the details of the container lifecycle, cleanup policy in k8s.

I think we need to do something a more cumbersome and link this file in to gradle home. We also should probably clean up the link before build completes so that the buildpack does not need to navigate possible broken links on subsequent builds.

Reply:
Yes, that's a great point. Don't want to put potentially sensitive info into the layer, even if it's just a cache/build layer. I pushed a change to switch to using symlinks. It doesn't delete the symlink when the buildpack finishes though, which ended up being kind of hard to test since the tests are looking to make sure the file is set up correctly. Instead, I made it such that it will handle the existence of a symlink & squash it (it's similar to what libbs is doing with the cache layer).